frontend/app/routes/_app+/new-application+/a2.tsx

```diff
@@ -0,0 +1,83 @@
+import { json, redirect } from '@remix-run/node';
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node';
+import { Form, useLoaderData } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+import { Button, ButtonLink } from '~/components/buttons';
+import { InputRadios } from '~/components/input-radios';
+import { getRaoidcService } from '~/services/raoidc-service.server';
+import { getSessionService } from '~/services/session-service.server';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('new-application');
+
+export const handle = {
+  breadcrumbs: [{ labelI18nKey: 'new-application:a2.breadcrumbs.new-application' }, { labelI18nKey: 'new-application:a2.breadcrumbs.a2' }],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-0016',
+  pageTitleI18nKey: 'new-application:a2.page-title',
+};
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const sessionService = await getSessionService();
+  const session = await sessionService.getSession(request);
+  const selectedType = session.get('selectedType');
+
+  return json({ selectedType });
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const sessionService = await getSessionService();
+  const session = await sessionService.getSession(request);
+
+  const formDataRadio = Object.fromEntries(await request.formData());
+  const selectedType = formDataRadio.selectedType === 'self';
+  session.set('selectedType', selectedType);
```

Comment:
Two things for this:
1. I don't think this needs to be stored server-side. Couldn't you just redirect to the correct next page depending on the selected value?
1. If it is required for some reason, the name of the variable doesn't really match the values.. `selectedType: true` and `selectedType: false`.

Reply:
1. I misread the mockup, you're right that doesn't need to be stored server side.
2. I've updated it.